custom_components/tahoma/tahoma_device.py

```diff
@@ -152,6 +152,14 @@ def has_state(self, *states: str) -> bool:
 
     async def async_execute_command(self, command_name: str, *args: Any):
         """Execute device command in async context."""
-        await self.coordinator.client.execute_command(
+        exec_id = await self.coordinator.client.execute_command(
             self.device.deviceurl, Command(command_name, list(args)), "Home Assistant"
         )
+
+        # ExecutionRegisteredEvent doesn't contain the deviceurl, thus we need to register it here
+        self.coordinator.executions[exec_id] = {
+            "deviceurl": self.device.deviceurl,
+            "command_name": command_name,
+        }
+
+        await self.coordinator.async_refresh()
```

Comment:
OK, I understand better the above code. Triggering an immediate update, you can catch the registered event.